src/app_service/service/wallet_connect/service.nim

```diff
@@ -202,3 +207,32 @@ QtObject:
       return ""
 
     return txResponse.getStr
+
+  # empty maxFeePerGasHex will fetch the current chain's maxFeePerGas
+  proc getEstimatedTimeMinutesInterval*(self: Service, chainId: int, maxFeePerGasHex: string): string =
+    var maxFeePerGas: float64
+    if maxFeePerGasHex.isEmptyOrWhitespace:
+      let chainFees = self.transactions.suggestedFees(chainId)
+      if chainFees.isNil:
+        return ""
+
+      # For non-EIP-1559 chains, we use the high fee
+      if chainFees.eip1559Enabled:
+        maxFeePerGas = chainFees.maxFeePerGasM
+      else:
+        maxFeePerGas = chainFees.maxFeePerGasL
+    else:
+      let maxFeePerGasInt = parseHexInt(maxFeePerGasHex)
+      maxFeePerGas = maxFeePerGasInt.float
+    let timeInterval = self.transactions.getEstimatedTime(chainId, $(maxFeePerGas))
+    case timeInterval
+    of EstimatedTime.Unknown:
+      return ""
+    of EstimatedTime.LessThanOneMin:
+      return "<1"
+    of EstimatedTime.LessThanThreeMins:
+      return "1-3"
+    of EstimatedTime.LessThanFiveMins:
+      return "3-5"
+    of EstimatedTime.MoreThanFiveMins:
+      return ">5"
```

Comment:
Pls don't return user visible strings from NIM or any backend code, those can't be translated

Reply:
Just return the enum value as `int` and compose the string in QML

Reply:
But those are not translatable, so I took a shortcut. I will follow on your recommendation and do it more structured; I was thinking the same but got lazy and applied the duct tape philosophy to go around nim's way of doing things :)

Reply:
You mean that ">5" doesn't show up in the UI in the end? I think it does

Reply:
> You mean that ">5" doesn't show up in the UI in the end? I think it does

I meant that this should work for all languages, or not? In the end I followed with you recommendation and used the helper function. It doesn't strictly follow the design requirements but at least it is consistent.

Reply:
No, not for those that don't use the Arabic numerals